documentation/src/docs/controller-service.md

```diff
@@ -794,23 +830,31 @@ other Controller instances will receive node failed notification and
+
+## KVT Operations
 
-The details about implementation, especially with respect to how the metadata is stored and managed is already discussed in the section [Cluster Listener](#controller-cluster-listener).
+Just as for streams, controller also runs all control plane operations for Key Value tables. This includes provisoning, sealing and deleting key value tables. These operations are also performed on the asynchronous event processing framework. A dedicated instance of EventProcessor is brought up for processing KV table requests which uses a new internal table-request-stream. Requests on this stream are distributed across different controller instances using kv table name as the routing key and each partition is spread across different controller instances. 
+
+### Create KVTable
+
+Create KVTable is modeled as a task on the event processor. When a request from user is received, a new event is posted into kvt-request-stream and picked asynchronously for  processing. As part of create KV table, the task creates the number of table segments specified in the KeyValueTablesConfiguration.numberOfPartitions and sets the state of the table as Active. 
+
+### Delete KVTable
+Delete KVTable is also modeled as a task on the event processor. When a request from user to delete the KV table is received, a new event is posted into kvt-request-stream and picked asynchronously for processing. As part of delete KV table, the task removes all table segments and all metadata that was created for the KV table. 
+
+### Readergroup Management
+With 0.9.0 we also introduced a concept of managing reader group configuration on controller. 
+A Pravega Reader Group is a group of readers that collectively read events from a Stream. A single segment in the Stream is assigned to exactly one reader in the ReaderGroup.
+A Reader Group has a configuration that describes the Streams it reads from and decides its checkpointing related behaviour. Controller service participates in readergroup's lifecycle from creation to resets to deletes where applications record changes to stream configuration with controller. Controller stores and manages the readergroup configuration and assigns versioning of readergroup config by introducing a new property called generation. Everytime readergroup is reset with a new configuration, this is recorded with controller and controller runs an asynchronous job to update the subscription status for streams that were added to or removed from the readergroup.
+To achieve this controller relies on event processor framework and has CreateReaderGroupTask, UpdateReaderGroupTask and DeleteReaderGroupTask which handle the subscription changes.  
+
+In addition controller also exposes apis using which readergroups can report their positions to controller. Controller Service tracks positions of these reported positions for multiple reader groups reading from a stream and takes them into consideration while deciding how much data to retain in the stream. 
```

Comment:
"Controller Service tracks positions of these reported positions" could be re-worded as:
"
The Controller maintains "read" positions for Reader-Groups with Consumption Based Retention and truncates the Stream based on a common lowest Stream-Cut across multiple such Reader Groups.
"

Reply:
since we have not explained what consumption based retention is, we shoud either explain it or pharse this statement without using the feature name.

Reply:
since we have not explained anywhere what consumption based retention is (and it is perhaps not required to be explained in this document) we could reword that suggested comment without its name.